debian/apport-general-hook.py

```diff
@@ -0,0 +1,62 @@
+import json
+import logging
+
+
+def get_ec2_data(instance_data):
+    """
+    provided cloud-init data from /run/cloud/instance-data.json
+    populate a smaller set of information helpful for debugging
+    """
+    stripped_data = {}
+    instance_identity_keys = (
+        "architecture",
+        "billingProducts",
+        "imageId",
+        "region",
+    )
+    for key in instance_identity_keys:
+        stripped_data[key] = instance_data["ds"]["dynamic"][
+            "instance-identity"
+        ]["document"][key]
+    stripped_data["instanceType"] = instance_data["ds"]["meta-data"][
+        "instance-type"
+    ]
+    return stripped_data
+
+
+PLATFORM_SPECIFIC_INFO = {"ec2": get_ec2_data}
```

Comment:
We may want to include Azure here too:
```suggestion
def get_azure_data(instance_data):
    azure_data = {}
    imds_name_to_report_map = {
        "publisher": "ImagePublisher",
        "offer": "ImageOffer",
        "sku": "ImageSKU",
        "version": "ImageVersion",
        "vmSize", "VMSize",
    }
    for src_key, report_name in imds_name_to_report_map.items():
        azure_data[report_key_name] = compute_data[src_key]
    return azure_data


PLATFORM_SPECIFIC_INFO = {"ec2": get_ec2_data, "azure": get_azure_data}
```

Example: instance data:https://hackmd.io/aAJFerDRRNaoL0Q_vDRYBg#Azure

some sample(a bit stale) cloud-init query output to give a glace at instance data formats per cloud.

Reply:
Added your get_azure_data and refactored it until it worked. Example response for the example data:
```python
{'ImagePublisher': 'Canonical',
 'ImageOffer': 'UbuntuServer',
 'ImageSKU': '19.04-DAILY',
 'ImageVersion': '19.04.201811290',
 'VMSize': 'Standard_DS1_v2'}
```